python/demo/demo_scipy-solvers.py

```diff
@@ -0,0 +1,148 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.13.6
+# ---
+# Copyright (C) 2024 Jørgen S. Dokken
+#
+# This file is part of DOLFINx (<https://www.fenicsproject.org>)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+#
+# # $L^2$ projection through minimization using SciPy solvers
+# In this demo, we will go through how to do a `projection` of a spatially varying function `g`,
+# into a function space `V`.
+#
+# An $L^2$ projection can be viewed as the following minimization problem:
+#
+# $$ \min_{u\in V} G(u) = \frac{1}{2}\int_\Omega (u-g)\cdot (u-g)~\mathrm{d}x.$$
+
+# We start by importing the necessary modules
+
+from mpi4py import MPI
+
+import numpy as np
+import numpy.typing as npt
+import scipy.optimize
+import scipy.sparse.linalg
+
+import dolfinx
+import ufl
+
+if MPI.COMM_WORLD.size > 1:
+    print("This demo works only in serial.")
```

Comment:
Would be good to say why, at least in a comment, because there is no issue until you use scipy sparse objects.

Reply:
Comment added above